algorithms/C/stacks/stack-using-linked-list.c

```diff
@@ -0,0 +1,182 @@
+// Implementation of stack using Linked List
+
+/*
+Implement the following stack operations : 
+1. Push 
+2. Pop 
+3. Peek
+4. Display
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+
+struct Stack
+{
+    int data;
+    struct Stack *next;
+};
+
+struct Stack *top = NULL;
+
+// Push Element at the top of stack
+void push()
+{
+    int x;
+    printf("Enter Element to be pushed\n");
+    scanf("%d", &x);
+    struct Stack *newNode = (struct Stack *)malloc(sizeof(struct Stack));
+    newNode->data = x;
+    newNode->next = top;
+    top = newNode;
+}
+
+// Pop the top-element from the stack
+void pop()
+{
+    struct Stack *temp = (struct Stack *)malloc(sizeof(struct Stack));
+    temp = top;
+    if (top == NULL)
+    {
+        printf("Stack is empty\n");
+    }
+    else
+    {
+        printf("Popped Element is %d\n", top->data);
+        top = top->next;
+        free(temp);
+    }
+}
+
+// Prints the top-most element
+void peek()
+{
+    if (top == NULL)
+    {
+        printf("Stack is empty\n");
+    }
+    else
+    {
+        printf("Top element is %d\n", top->data);
+    }
+}
+
+// Prints The Stack
+void display()
+{
+    struct Stack *temp = (struct Stack *)malloc(sizeof(struct Stack));
+    temp = top;
```

Comment:
```suggestion
    struct Stack *temp = top;
```

memory allocated to `temp` in previous statement will leak here. 
btw I fail to understand the purpose of allocating memory to `temp` at first place.